.github/workflows/ci.yml

```diff
@@ -0,0 +1,149 @@
+name: CI
+
+env:
+  go-version: '1.14.6'
+
+# Controls when the action will run. Triggers the workflow on pull requests to a set of branches.
+on:
+  pull_request:
+    branches: [
+      # This is our working branch
+      develop,
+      # This is where pull requests from "bors r+" are built.
+      staging,
+      # This is where pull requests from "bors try" are built. 
+      trying,
+    ]
+  push:
+    branches: [
+      develop,
+      staging,
+      trying,
+    ]
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # no freebsd support for now =\
+        #os: [macos-latest, windows-latest, ubuntu-latest]
+        os: [ubuntu-latest]
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+    - name: set up go
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.go-version }}
+    - name: test build
+      run: make
+    - name: unit tests (except app test)
+      run: make test-no-app-test
+    - name: app test
+      run: make test-only-app-test
+    - name: fmt, tidy, lint
+      # lint gets confused by windows line endings so skip it on windows
+      if: matrix.os != 'windows-latest'
+      run: |
+        make genproto
+        make test-tidy
+        make test-fmt
+        make lint
+    - name: push to dockerhub
+      # only run this once, and only on push
+      if: github.event == 'push' && matrix.os == 'ubuntu-latest'
+      run: make dockerpush
+
+  # run system tests as independent jobs so they can run in parallel
+  systemtest-latenodes:
+    needs: test
+    if: github.event == 'push' && (github.ref == 'staging' || github.ref == 'trying')
+    runs-on: ${{ matrix.os }}
```

Comment:
This should never run on more than one platform.

Reply:
You're referring specifically to the `latenodes` test? Out of curiosity, why only this one?